src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/RemoteDiagnosticAnalyzerService.cs

```diff
@@ -98,5 +100,28 @@ public ValueTask ReportAnalyzerPerformanceAsync(ImmutableArray<AnalyzerPerforman
                 return default;
             }, cancellationToken);
         }
+
+        public ValueTask<bool> HasRefactoringsAsync(PinnedSolutionInfo solutionInfo, DocumentId documentId, TextSpan textSpan, TextSpan? pastedTextSpan, CancellationToken cancellationToken)
```

Comment:
there's a potential future optimizatino we can do/track if we're willing to make the statement that a refactoring should be computable without knowing about projects that depend on.  Specifically, OOP supports the idea of quickly syncing only a project (and the projects it depends on).  we could use that to more quickly sync and find out if there are refactorings.  i believe all *our* refactorings would satisfy this.  but some third party ones might not.